src/storage/storage.cpp

```diff
@@ -184,643 +182,670 @@ Storage::ReturnCode Storage::Run(int max_wait)
+            layout_ptr->GetBlockPtr<unsigned, true>(memory_ptr, DataLayout::GEOMETRIES_INDEX);
+        BOOST_ASSERT(geometry_index_count == layout_ptr->num_entries[DataLayout::GEOMETRIES_INDEX]);
+        geometry_input_file.readInto(geometries_index_ptr, geometry_index_count);
 
-    // store timestamp
-    char *timestamp_ptr =
-        shared_layout_ptr->GetBlockPtr<char, true>(shared_memory_ptr, SharedDataLayout::TIMESTAMP);
-    io::readTimestamp(timestamp_stream, timestamp_ptr, timestamp_size);
+        const auto geometries_node_id_list_ptr =
+            layout_ptr->GetBlockPtr<NodeID, true>(memory_ptr, DataLayout::GEOMETRIES_NODE_LIST);
+        const auto geometry_node_lists_count = geometry_input_file.readElementCount32();
+        BOOST_ASSERT(geometry_node_lists_count ==
+                     layout_ptr->num_entries[DataLayout::GEOMETRIES_NODE_LIST]);
+        geometry_input_file.readInto(geometries_node_id_list_ptr, geometry_node_lists_count);
 
-    // store search tree portion of rtree
-    RTreeNode *rtree_ptrtest = shared_layout_ptr->GetBlockPtr<RTreeNode, true>(
-        shared_memory_ptr, SharedDataLayout::R_SEARCH_TREE);
-    io::readRamIndex(tree_node_file, rtree_ptrtest, tree_size);
-
-    // load core markers
-    std::vector<char> unpacked_core_markers(number_of_core_markers);
-    core_marker_file.read((char *)unpacked_core_markers.data(),
-                          sizeof(char) * number_of_core_markers);
+        const auto geometries_fwd_weight_list_ptr = layout_ptr->GetBlockPtr<EdgeWeight, true>(
+            memory_ptr, DataLayout::GEOMETRIES_FWD_WEIGHT_LIST);
+        BOOST_ASSERT(geometry_node_lists_count ==
+                     layout_ptr->num_entries[DataLayout::GEOMETRIES_FWD_WEIGHT_LIST]);
+        geometry_input_file.readInto(geometries_fwd_weight_list_ptr, geometry_node_lists_count);
 
-    unsigned *core_marker_ptr = shared_layout_ptr->GetBlockPtr<unsigned, true>(
-        shared_memory_ptr, SharedDataLayout::CORE_MARKER);
+        const auto geometries_rev_weight_list_ptr = layout_ptr->GetBlockPtr<EdgeWeight, true>(
+            memory_ptr, DataLayout::GEOMETRIES_REV_WEIGHT_LIST);
+        BOOST_ASSERT(geometry_node_lists_count ==
+                     layout_ptr->num_entries[DataLayout::GEOMETRIES_REV_WEIGHT_LIST]);
+        geometry_input_file.readInto(geometries_rev_weight_list_ptr, geometry_node_lists_count);
+    }
 
-    for (auto i = 0u; i < number_of_core_markers; ++i)
     {
-        BOOST_ASSERT(unpacked_core_markers[i] == 0 || unpacked_core_markers[i] == 1);
+        io::File geometry_datasource_file(config.datasource_indexes_path);
+        const auto number_of_compressed_datasources = geometry_datasource_file.readElementCount64();
 
-        if (unpacked_core_markers[i] == 1)
+        // load datasource information (if it exists)
+        const auto datasources_list_ptr =
+            layout_ptr->GetBlockPtr<uint8_t, true>(memory_ptr, DataLayout::DATASOURCES_LIST);
```

Comment:
Related: can we change the `bool` here to an enum? What is `true`? (I know shm but still this is semantically meaningless in this context)